### PR TITLE
[CI-7143] [DOC-2640] - Bitbucket connector settings reference

### PR DESCRIPTION
**Pull request by aimurphy**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [x] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* CI-7143: Clarify URL format for **Bitbucket Account/Repository URL** field. Added more explanation, examples, and screenshots.
* DOC-2640: Added a screenshot clarifying where the account username comes from, in case the user has changed the display name.
* General cleanup & style changes
* Set sidebar order

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [x] *Optional* Screen Shoot.

### CODE REVIEW

**Comment by devkimittal** [comment]
@aimurphy Looks good to me, just FYI bitbucket onprem url can also look like abcd.com, ab.cd.com, a.b.cd.com

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 1]
```diff
@@ -1,98 +1,209 @@
...
+
+The **HTTP** Connection Type requires **Username** and **Password** authentication for all accounts and repos, including read-only repos.
+
+In the **Username** field, enter the Bitbucket account username as specified in your Bitbucket **Account settings**.
+
+![Bitbucket Personal settings screen, highlighting the Account settings page and the Username field.](./static/bitbucket-username-in-acct-settings.png)
+
+In the **Password** field, provide a Bitbucket [Access token](https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/), [HTTP access token](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html), or [App password](https://support.atlassian.com/bitbucket-cloud/docs/create-an-app-password/). Passwords are stored as [Harness Encrypted Text secrets](../../6_Security/2-add-use-text-secrets.md).
+
+App passwords need the following permissions:
```
Comment (devkimittal) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:139]: I feel this is not necessary. KInda misguiding as well.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 2]
```diff
@@ -1,98 +1,209 @@
...
+  <TabItem2 value="http" label="Username and Password" default>
+```
+
+The **HTTP** Connection Type requires **Username** and **Password** authentication for all accounts and repos, including read-only repos.
+
+In the **Username** field, enter the Bitbucket account username as specified in your Bitbucket **Account settings**.
+
+![Bitbucket Personal settings screen, highlighting the Account settings page and the Username field.](./static/bitbucket-username-in-acct-settings.png)
+
+In the **Password** field, provide a Bitbucket [Access token](https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/), [HTTP access token](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html), or [App password](https://support.atlassian.com/bitbucket-cloud/docs/create-an-app-password/). Passwords are stored as [Harness Encrypted Text secrets](../../6_Security/2-add-use-text-secrets.md).
```
Comment (devkimittal) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:137]: We can add a note here saying only account level token/passwords we accept. Some bitbucket servers also have repo level tokens, connection test will fail if they use repo level.

**Comment by aimurphy** [comment]
> 

I added more detail to the `:::tip` about different posiblities for onprem URLs.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md (hunk 1 above)
Comment (aimurphy) [reply docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:139]: removed the permission list and added a general troubleshooting advice.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md (hunk 2 above)
Comment (aimurphy) [reply docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:137]: Added in both places where tokens/app passwords are mentioned.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 3]
```diff
@@ -1,98 +1,201 @@
...
-* **HTTPS:** Requires a personal access token.
-* **SSH:** You must use a key in PEM format, not OpenSSH.  
-To generate an SSHv2 key, use: `ssh-keygen -t rsa -m PEM` The `rsa` and `-m PEM` ensure the algorithm and that the key is PEM. Next, follow the prompts to create the PEM key. For more information, see the [ssh-keygen man page](https://linux.die.net/man/1/ssh-keygen).
+![Bitbucket connector Details settings configured to connect to an On-Prep account using an SSH URL with a port number.](./static/bitbucket-connector-settings-reference-ssh-with-port.png)
 
-### Bitbucket Account URL
+:::
 
-The URL for your Git repo. Make sure that it matches the Connection Type option you selected.
+### Test Repository
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:102]: ### Test repository

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 4]
```diff
@@ -1,98 +1,201 @@
...
+:::caution
 
-#### Enable API access
+For **HTTP** Connection Types, use the same password you used earlier, and make sure the **Username** fields are both plaintext or encrypted. You can't use a plain-text username for one field and a secret for the other.
 
-This option is required for using Git-based triggers, Webhook management, and updating Git statuses. If you are using Harness Git Experience, you will need to use this setting.
+:::
 
-### API Authentication
+## Connectivity Mode settings
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:190]: Connectivity mode settings

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 5]
```diff
@@ -1,98 +1,201 @@
...
+
+## Credentials settings
+
+Provide authentication credentials for the connector.
 
 ### Authentication
 
-Bitbucket repos with read-only access also require a username and password.You can use a password for HTTPS credentials.
+The **Authentication** method is determined by the **Connection Type** you chose in the [Details settings](#details-settings).
+
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:120]: The **Connection Type** you select in the [Details settings](#details-settings) determines the **Authentication** method.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 6]
```diff
@@ -1,98 +1,201 @@
...
+  <TabItem2 value="http" label="Username and Password" default>
+```
+
+The **HTTP** Connection Type requires **Username** and **Password** authentication for all accounts and repos, including read-only repos.
+
+In the **Username** field, enter the Bitbucket account username as specified in your Bitbucket **Account settings**.
+
+![Bitbucket Personal settings screen, highlighting the Account settings page and the Username field.](./static/bitbucket-username-in-acct-settings.png)
+
+In the **Password** field, provide a Bitbucket account-level [Access token](https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/), [HTTP access token](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html), or [App password](https://support.atlassian.com/bitbucket-cloud/docs/create-an-app-password/). Passwords are stored as [Harness Encrypted Text secrets](../../6_Security/2-add-use-text-secrets.md).
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:137]: ... provide a Bitbucket account-level [access token]

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 7]
```diff
@@ -1,98 +1,201 @@
...
+
+You must provide an account-level app password or token. Repo-level tokens are not supported.
+
+If you use a Google account to log in to Bitbucket, you must use an App password.
+
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:142]: Is it app password or App password? Can we say application password instead?

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 8]
```diff
@@ -1,98 +1,201 @@
...
+```mdx-code-block
+  </TabItem2>
+  <TabItem2 value="ssh" label="SSH Key">
+```
+
+The **SSH** Connection Type requires an **SSH Key** in PEM format. OpenSSH keys are not supported. In Harness, SSH keys are stored as [Harness Encrypted File secrets](../../6_Security/3-add-file-secrets.md).
+
+For details on creating SSH keys and adding them to your Bitbucket account, go to the Bitbucket documentation to [Configure SSH and two-step verification](https://support.atlassian.com/bitbucket-cloud/docs/configure-ssh-and-two-step-verification/).
+
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:153]: For details on creating SSH keys and adding them to your Bitbucket account, go to the Bitbucket documentation to [Configure SSH and use two-step verification](https://support.atlassian.com/bitbucket-cloud/docs/configure-ssh-and-two-step-verification/).

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 9]
```diff
@@ -1,98 +1,201 @@
...
+:::tip
+
+You use arguments such as `rsa` and `-m PEM` in your `keygen` commands to ensure your key is properly formatted and uses the RSA algorithm, for example this command creates an SSHv2 key:
+
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:157]: Use arguments such as `rsa` and `-m PEM` in your `keygen` commands to ensure your key is properly formatted and uses the RSA algorithm. For example, this command creates an SSHv2 key:

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 10]
```diff
@@ -1,98 +1,201 @@
...
+In the **Personal Access Token** field, provide a Bitbucket account-level [App password](https://support.atlassian.com/bitbucket-cloud/docs/create-an-app-password/), [Access token](https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/), or [HTTP access token](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html). Passwords are stored as [Harness Encrypted Text secrets](../../6_Security/2-add-use-text-secrets.md).
 
-#### SSH Key
+You must provide an account-level app password or token. Repo-level tokens are not supported.
 
-If you selected **SSH** as the connection protocol, you must add the **SSH Key** for use with the connection as a [Harness Encrypted File secret](../../6_Security/3-add-file-secrets.md). For steps to create an SSH Key, see [Add new SSH Key](https://support.atlassian.com/bitbucket-cloud/docs/set-up-an-ssh-key/).
+:::caution
 
-#### Enable API access
+For **HTTP** Connection Types, use the same password you used earlier, and make sure the **Username** fields are both plaintext or encrypted. You can't use a plain-text username for one field and a secret for the other.
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:186]: For **HTTP** Connection Types, use the same password you used earlier, and make sure the **Username** fields are both plaintext and encrypted. You can't use a plaintext username for one field and a secret for the other.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 11]
```diff
@@ -1,98 +1,201 @@
...
 
-This option is required for using Git-based triggers, Webhook management, and updating Git statuses. If you are using Harness Git Experience, you will need to use this setting.
+:::
 
-### API Authentication
+## Connectivity Mode settings
 
-#### UserName
+Select whether you want Harness to connect directly to your Bitbucket account/repo, or if you want Harness to communicate with your Bitbucket account/repo through a delegate.
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:192]: Select whether you want Harness to connect directly to your Bitbucket account or repository, or if you want Harness to communicate with your Bitbucket account or repository through a delegate.

File: docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md [hunk 12]
```diff
@@ -1,98 +1,201 @@
...
 
-The username for the account.
+## Troubleshooting
 
-You must enter a plain-text username or a username secret for *both* Authentication and API Authentication. You cannot use a plain-text password for one field and a secret for the other.
+If the connection test returns a `not authorized` error, make sure you used the **Username** shown in the Bitbucket **Account settings**.
```
Comment (SmritiSatya) [inline docs/platform/7_Connectors/ref-source-repo-provider/bitbucket-connector-settings-reference.md:196]: ..., ensure you used the ...

**Review by SmritiSatya — APPROVED** [review]

**Pull request merged**